// cypress-mongodb/dist/commands/collection.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dropCollection = exports.createCollection = void 0;
var validator_1 = require("../utils/validator");
function createCollection(collection, options) {
    var _a;
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: collection,
            failSilently: (_a = options === null || options === void 0 ? void 0 : options.failSilently) !== null && _a !== void 0 ? _a : false,
        },
    };
    (0, validator_1.validate)(args);
    return cy.task('createCollection', args).then(function (result) {
        return result;
    });
}
exports.createCollection = createCollection;
function dropCollection(collection, options) {
    var _a;
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: collection,
            failSilently: (_a = options === null || options === void 0 ? void 0 : options.failSilently) !== null && _a !== void 0 ? _a : false,
        },
    };
    (0, validator_1.validate)(args);
    return cy.task('dropCollection', args).then(function (result) {
        return result;
    });
}
exports.dropCollection = dropCollection;
