// cypress-mongodb/dist/commands/delete.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteMany = exports.deleteOne = void 0;
var validator_1 = require("../utils/validator");
function deleteOne(filter, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: (options === null || options === void 0 ? void 0 : options.collection) || Cypress.env('mongodb').collection,
        },
        pipeline: filter,
    };
    (0, validator_1.validate)(args);
    if (!filter) {
        throw new Error('Filter must be specified');
    }
    else if (typeof filter !== 'object' || Array.isArray(filter)) {
        throw new Error('Filter must be an object');
    }
    return cy.task('deleteOne', args).then(function (result) {
        return result;
    });
}
exports.deleteOne = deleteOne;
function deleteMany(filter, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: (options === null || options === void 0 ? void 0 : options.collection) || Cypress.env('mongodb').collection,
        },
        pipeline: filter,
    };
    (0, validator_1.validate)(args);
    if (!filter) {
        throw new Error('Filter must be specified');
    }
    else if (typeof filter !== 'object' || Array.isArray(filter)) {
        throw new Error('Filter must be an object');
    }
    return cy.task('deleteMany', args).then(function (result) {
        return result;
    });
}
exports.deleteMany = deleteMany;
