// cypress-mongodb/dist/commands/find.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.findMany = exports.findOne = void 0;
var validator_1 = require("../utils/validator");
var bson_1 = require("bson");
function findOne(query, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: (options === null || options === void 0 ? void 0 : options.collection) || Cypress.env('mongodb').collection,
        },
        pipeline: query,
    };
    (0, validator_1.validate)(args);
    if (!query) {
        throw new Error('Query must be specified');
    }
    else if (typeof query !== 'object' || Array.isArray(query)) {
        throw new Error('Query must be a valid mongodb query object');
    }
    return cy.task('findOne', args).then(function (result) {
        if (result !== null)
            return (0, bson_1.deserialize)(Buffer.from(result));
        else
            return null;
    });
}
exports.findOne = findOne;
function findMany(query, options) {
    var args = {
        uri: Cypress.env('mongodb').uri,
        options: {
            database: (options === null || options === void 0 ? void 0 : options.database) || Cypress.env('mongodb').database,
            collection: (options === null || options === void 0 ? void 0 : options.collection) || Cypress.env('mongodb').collection,
        },
        pipeline: query,
    };
    (0, validator_1.validate)(args);
    if (!query) {
        throw new Error('Query must be specified');
    }
    else if (typeof query !== 'object' || Array.isArray(query)) {
        throw new Error('Query must be a valid mongodb query object');
    }
    return cy.task('findMany', args).then(function (result) {
        return Object.values((0, bson_1.deserialize)(Buffer.from(result)));
    });
}
exports.findMany = findMany;
